hw/ip_templates/rv_core_ibex/rtl/rv_core_ibex.sv.tpl

```diff
@@ -608,7 +628,7 @@ module ${module_instance_name}
   ) tl_adapter_host_i_ibex (
     .clk_i,
     .rst_ni,
-    .req_i        (instr_req),
+    .req_i        (instr_req_q),
     .instr_type_i (prim_mubi_pkg::MuBi4True),
     .gnt_o        (instr_gnt),
```

Comment:
It seems the tests are failing, and I think the reason might be the following: Ibex's `instr_req` signal and the corresponding data is now delayed one cycle, but the `instr_gnt` signal is going the other direction. So at Ibex's interface, in the cycle it sees a `req`/`gnt` handshake taking place and then releases its `req` signal, the `tl_adapter_host` here will still see the `req` signal asserted one cycle later and might register a second handshake for the same request? Or in a different scenario, if `gnt` is high by default and pulled low if a request needs to be stalled, then Ibex could see a handshake, while the adapter doesn't.

So unless I am misunderstanding something here, in such a handshake protocol, cutting the path usually has to take into account also the `gnt`.

Reply:
Thanks I saw the change is failing an created traces in the last hour. Let me confirm the behavior.